app/src/actions/vesselInfo.js

```diff
@@ -64,11 +64,10 @@ function setCurrentVessel(tilesetId, seriesgroup, fromSearch) {
       if (request.readyState !== 4) {
         return;
       }
-      if (request.status >= 500) {
+      if (request.status >= 500 || request.responseText === 'Not Found') {
```

Comment:
if the response text is "Not Found", we should get a matching http code, and test for that.